### ISSUE TITLE
clean up orphaned branches

### ISSUE THREAD

**Issue by MathewBiddle** [ioos/documentation-theme-jekyll#3]
can we clean up some of the non-relevant branches in this repo? 

I'm getting confused as to which ones are used for our documentation theme and which are dev.

**Comment by MathewBiddle** [comment]
I'm curious about the approach with the navbars being a branch on this repo instead of it's own repo. Was the just to keep the theme stuff together in one repo? 

For me, having the navbars as a branch on this repo makes it very confusing and tricky to work with submodules. If it's of interest, I can propose how we would pull the navbars branch into another repo and use that, instead of the branch here.

**Comment by mwengren** [comment]
Hey @MathewBiddle there's definitely a lot that can be cleaned up with this repo and the documentation theme overall.  I think we can lose the branches that came with the original code for sure, those were in the source repo when it was initially forked.  I see three of those: reviews, docreviews, and revert-24-gh-pages.  

As far as the branches and submodules, that's why that approach was used: to keep everything together in one repo.  I don't think we could think of a better to manage everything via submodules, so that's what we went with.  The purpose for using submodules was to be able to update the theme in one location (master branch of this repo - plus one other branch containing the topnav config file because they couldn't be the same AFAIK - rather than make the change x # of times).

I guess my preference would still be to keep everything together in one repo and keep the topnav branch as a branch - hopefully simplified (see below).

We probably should have named the `navbars` branch in a more clear way, but maybe we can do that now?

From your issue in the skeleton repo: https://github.com/ioos/ioos-documentation-jekyll-skeleton/issues/4, the 'master' branch content is pulled into each documentation repo in the `documentation-theme-jekyll` subdirectory:
```
git submodule add --name documentation-theme-jekyll https://github.com/ioos/documentation-theme-jekyll.git theme
```
And the `navbars` branch is pulled into the `_data/navbars_theme` subdirectory (I guess that's why it was named as it was, now that I see it again):
```
git submodule add --name _data/navbars_theme https://github.com/ioos/documentation-theme-jekyll _data/navbars_theme
git submodule set-branch --branch navbars _data/navbars_theme
```
Maybe we could rename that `submodule-topnav` perhaps?  And the directory to _data/topnav instead (especially if what I said below about only needing the `topnav_ioos.yml` file is true.

It gets tricky because the theme itself is configured to read yaml files from certain directories for configuration - those are all set in the `_config.yml` file, but in the case of the ioos-documentation-jekyll-skeleton repo (and all of our doc repos descended from it), that's this _config.yml: https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/gh-pages/_config.yml,  not from master branch in this repo.  The directory paths at the [bottom](https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/gh-pages/_config.yml#L153-L158) are relative to the root of the repo itself (not the submodule with the theme code, if that makes sense).

To be honest, I don't remember if we actually need any of the other files in the `navbars_theme` branch other than `topnav_ioos.yml`, because they're actually duplicated both [here](https://github.com/ioos/ioos-documentation-jekyll-skeleton/tree/gh-pages/_data) and [here](https://github.com/ioos/documentation-theme-jekyll/tree/248d4c8ac0091d4f36b92ee39e45903e9fcd6d8e) in the skeleton repo.  I'm not sure which of the other duplicated yml config files are used in the end (alerts.yml, glossary.yml, etc).

You could drop all the other files from that branch and see if it still works, that would clean things up a bit.

Unfortunately, it will always be a bit of a mess that's hard to maintain, but there's room for improvement.

**Comment by mwengren** [comment]
@MathewBiddle this may be overkill, because we'll have to repeat it x number of times for each existing repo, plus the https://github.com/ioos/ioos-documentation-jekyll-skeleton repo itself, but in addition to removing extra yml files in the [navbars](https://github.com/ioos/documentation-theme-jekyll/tree/navbars) submodule branch if they prove redundant, we could also remove the local [topnav.yml](https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/gh-pages/_data/topnav.yml) (redundant with topnav_ioos.yml in the navbars submodule)

The other possibility for that collection of YAML config files in the `navbars` theme is that Jekyll is actually reading them instead from the [_data](https://github.com/ioos/ioos-documentation-jekyll-skeleton/tree/gh-pages/_data) directory, in which case they are unique to each repo, similar to [sidebar_ioos.yml](https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/gh-pages/_data/sidebars/sidebar_ioos.yml).  In that case I guess it's best left alone.  

Let me know if you want me to help with any of this...

**Comment by MathewBiddle** [comment]
Looks like we can safely remove the following from [`navbars`](https://github.com/ioos/documentation-theme-jekyll/tree/navbars) (essentially only leaving `topnav_ioos.yml`):

```
            alerts.yml
            definitions.yml
            glossary.yml
            home_sidebar.yml
            mydoc_sidebar.yml
            other_sidebar.yml
            product1_sidebar.yml
            product2_sidebar.yml
            samplelist.yml
            strings.yml
            tags.yml
            terms.yml
            topnav.yml
```


As for [`master`](https://github.com/ioos/documentation-theme-jekyll/tree/master), I think you can remove the following without issue (at least nothing broke when I rendered mbon-docs locally without these files):
```
            .vscode/settings.json
            _data/alerts.yml
            _data/definitions.yml
            _data/glossary.yml
            _data/samplelist.yml
            _data/sidebars/home_sidebar.yml
            _data/sidebars/mydoc_sidebar.yml
            _data/sidebars/other.yml
            _data/sidebars/product1_sidebar.yml
            _data/sidebars/product2_sidebar.yml
            _data/strings.yml
            _data/tags.yml
            _data/terms.yml
            _data/topnav.yml
            licenses/LICENSE
            licenses/LICENSE-BSD-NAVGOCO.txt
            pdfconfigs/config_mydoc_pdf.yml
            pdfconfigs/config_product1_pdf.yml
            pdfconfigs/config_product2_pdf.yml
            pdfconfigs/prince-list.txt
            pdfconfigs/titlepage.html
            pdfconfigs/tocpage.html
```

In the repo of your doc site (in this case [mbon-docs](https://github.com/ioos/mbon-docs)), you can safely remove the following files from [_data](https://github.com/ioos/mbon-docs/tree/gh-pages/_data):
```
            _data/alerts.yml
            _data/definitions.yml
            _data/glossary.yml
            _data/samplelist.yml
            _data/strings.yml
            _data/terms.yml
            _data/topnav.yml
```

You can probably get rid of some of the root files in [`master`](https://github.com/ioos/documentation-theme-jekyll/tree/master) as well. eg. Gemfile, Dockerfile, shell scripts, etc. I'll dig into that next.

**Comment by MathewBiddle** [comment]
by removing `_data/alerts.yml` from [mbon_docs](https://github.com/ioos/mbon-docs/blob/gh-pages/_data/alerts.yml) we lost the shaded line at the end of each post.

w/ shaded line: ![image](https://user-images.githubusercontent.com/8480023/218558212-5bf63f49-08dd-4b2f-a02e-6856ea51bfcc.png)

w/out shaded line:
![image](https://user-images.githubusercontent.com/8480023/218559125-a734775c-f9ed-47ff-a4ff-4de9e4e8a6de.png)



This is invoked at https://github.com/ioos/documentation-theme-jekyll/blob/6674af5f1b065d2d24e358472f428f864b023c60/_layouts/post.html#L37

If we don't care about that line, we can keep `_data/alerts.yml` out of the doc site. Otherwise, we keep it in there. 

Here is the line which adds the shaded line to a post. 
https://github.com/ioos/mbon-docs/blob/09f5261b9d36fc267a9a0f5ae101dd0877fe8a5d/_data/alerts.yml#L15

**Comment by MathewBiddle** [comment]
IMO, we should leave the config .yml files in `_data/` for the individual doc repos. 

We should remove the files above out of [master](https://github.com/ioos/documentation-theme-jekyll/tree/master) and [navbars](https://github.com/ioos/documentation-theme-jekyll/tree/navbars) as they are not being used even if they exist. The only ones that can be read are the root documentation `_data/` as well as `_data/sidebars/` and `_data/navbars_theme/`. 

This cleans up the [documentation-theme-jekyll](https://github.com/ioos/documentation-theme-jekyll) repo as a start.

**Comment by mwengren** [comment]
@MathewBiddle I am +1 for removing the files in the theme.  I see your PRs so I guess I'll just go there instead.  Maybe not sure about the `/pdfconfigs` and `/licenses` folders tho.  The PDF stuff I think allows you to output all the docs in a PDF rather than HTML via Jeklyy, so it might be useful still.   

I suspect we kept all of these files in initially to keep in line with the upstream repo, but realistically we're probably never going to pull in any changes from there anyway.  It would be better to start over re-fork and repeat all of the styling changes I did early on to make it ioos.us-like.  So, in favor of simplifying for easier maintenance.

Also, agree on keeping the individual files in downstream documentation repos in the `_docs` dir if needed in the future.  That's basically where site-specific config elements belong.

There are other duplicated files in each doc repo and this one (`_config.yml` and `Gemfile`) but some of those might be useful to keep as a reference, IDK.  There's only a few anyway.

**Comment by mwengren** [comment]
Sorry, another thought, maybe we should take this opportunity to change master to main for this repo and update the various submodule configs.  Difficult?  Hopefully not...

The branch name change would have to happen first of course.  I'm going to make changes in https://github.com/ioos/ioos-documentation-jekyll-skeleton to match what you did in https://github.com/ioos/mbon-docs wrt GHA.

I'd also like to rename `_data/navbars_theme` to `_data/navbars`, but I worry there's a path config somewhere with that path hardcoded.  But maybe not...

If we're going to change the master branch to main, might as well make the above path name change as well since both require modifications to `.gitmodules' (and hopefully nothing else).

**Comment by MathewBiddle** [comment]
I had a similar thought on renaming the branch. We would need to update .gitmodules for each of the doc sites, but that's fairly simple. In fact, I think we might not need to do that if submodules defaults to use the default branch.

From the [docs](https://git-scm.com/docs/git-submodule):
>The default remote is the remote of the remote-tracking branch of the current branch. If no such remote-tracking branch exists or the HEAD is detached, "origin" is assumed to be the default remote. If the superproject doesn’t have a default remote configured the superproject is its own authoritative upstream and the current working directory is used instead.

And these [docs](https://git-scm.com/docs/gitmodules#Documentation/gitmodules.txt-submoduleltnamegtbranch):
> If the [branch] option is not specified, it defaults to the remote HEAD

When you say rename `_data/navbars_theme`, do you mean the path or the name for the submodule?
```
[submodule "_data/navbars_theme"]
	path = _data/navbars_theme
	url = https://github.com/ioos/documentation-theme-jekyll
	branch = navbars
```

At the end of the day, I don't think either of those matter for Jekyll. After doing some [reading](https://jekyllrb.com/docs/datafiles/), as long as the `navbars` submodule is nested under `_data/`, the `.yml` files will be picked up and read appropriately. 

We do explicitly call `topnav_ioos` in `_config.yml`, but we aren't planning on changing the filename, so I think we're okay. 

TLDR; I say we make those changes!

**Comment by mwengren** [comment]
I looked at the submodule rename last night as well and I think you're right the master -> main branch change name shouldn't affect anything in the downstream `.gitmodules` file configurations, so we can do that at any point.

As far as the `navbars_theme`, I just didn't like the path name that was used, `_data/navbars` seems more intuitive.  

Turns out that's easy to change in some testing I did in the skeleton repo, only involves a git mv and config files are updated accordingly - I'll submit another PR for that shortly:
```
git mv _data/navbars_theme _data/navbars
```
.gitmodules:
```
[submodule "_data/navbars_theme"]
	path = _data/navbars
	url = https://github.com/ioos/documentation-theme-jekyll
	branch = navbars
```
Changing the alias or name of the submodule itself `[submodule "_data/navbars_theme"]` seems more difficult and not worth the effort anyway.  The same alias is also used in `.git/config` file:
```
[submodule "_data/navbars_theme"]
	url = https://github.com/ioos/documentation-theme-jekyll
```
Modifying those manually or deleting and recreating the `navbars` branch submodule only to change the alias name seems like overkill.

So, if we just add the git mv command to an update process for each doc repo, we should be good.

**Comment by MathewBiddle** [comment]
Agreed on the overkill w/ renaming the alias/name.

**Comment by mwengren** [comment]
Well, strangely, now it's not working when I test locally.  The 'IOOS Documentation & Data Sites' pulldown is dropped from the menu bar.  

![1676478490000](https://user-images.githubusercontent.com/1469921/219090530-8a690aee-a510-414d-8c30-1c31974d1906.png)



I pushed the branch [here](https://github.com/mwengren/ioos-documentation-jekyll-skeleton/tree/submodule-rename-2) but didn't submit a PR yet.  

If you can test locally that might help.  I also pushed it to the main repo as well [here](https://github.com/ioos/ioos-documentation-jekyll-skeleton/tree/submodule-rename-2) so you don't need to bother with my fork.  Thanks!

**Comment by MathewBiddle** [comment]
Interesting - works fine for me. I checked out https://github.com/ioos/ioos-documentation-jekyll-skeleton/tree/submodule-rename-2 and this is what I render:

![image](https://user-images.githubusercontent.com/8480023/219114797-25d9925b-8caa-45ef-833c-61685ae95e1a.png)

Maybe try `bundle exec jekyll clean` and then redeploy?

**Comment by MathewBiddle** [comment]
standby - I didn't pull the submodules in. Now I'm getting the same issue.

**Comment by MathewBiddle** [comment]
Found it!!

In the [theme](https://github.com/ioos/documentation-theme-jekyll) (`master` branch) we are explicitly calling to `site.data.navbars_theme[page.topnav].topnav_dropdowns`, note the `navbars_theme` piece. Adjusting this to `navbars` resolves the issue.

https://github.com/ioos/documentation-theme-jekyll/blob/6674af5f1b065d2d24e358472f428f864b023c60/_includes/topnav.html#L28

So, we need to do two things.
1. update the alias for the submodule in all of the repos
2. update that line in `master` to reflect the new alias.

**Comment by MathewBiddle** [comment]
For reference, if we wanted to change `sidebars` we'd need to update https://github.com/ioos/documentation-theme-jekyll/blob/6674af5f1b065d2d24e358472f428f864b023c60/_includes/sidebar.html#L1 similarly.

**Comment by mwengren** [comment]
Crap, ok.  I'll look at it shortly in my branch, along with some other changes.  Thanks!

**Comment by mwengren** [comment]
Well, at least the last time I looked at that was [~4 years ago](https://github.com/ioos/documentation-theme-jekyll/commit/db5b4401c64cf4e2a88559453a4af292afefea66), and not more recently, so that makes me feel a little better.

I'll just push a new commit with that change, rather than a PR.  

Fortunately, it shouldn't affect any of the existing sites until they pull in the submodule theme updates, so we can plan ahead to make the `navbars` branch path change, add GHA job to update at the same time, and hopefully each repo will pull in the submodule update at that time without breaking anything already deployed.

I also noticed in most of the repos probably we have a redundant `_data/topnav.yml` file that can be removed simultaneously.   So my list of per-repo changes looks something like:

> M .gitignore  - see [this commit](https://github.com/ioos/ioos-documentation-jekyll-skeleton/commit/fdac97bfa343b1603e132cd75c233ddc21ed7b06#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947)
> M _includes/topnav.yml - [line 28](https://github.com/ioos/documentation-theme-jekyll/blob/6674af5f1b065d2d24e358472f428f864b023c60/_includes/topnav.html#L28)
> D  _data/topnav.yml
> A .github/workflows/sync_theme.yml [here](https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/gh-pages/.github/workflows/sync_theme.yml)
> M .gitmodules [line 5](https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/dd81c9e03ae9272b7c2e4bf42107b29e956db6b9/.gitmodules#L5)
>
> git mv _data/navbars_theme _data/navbars

Or something like that.   If there's something else missing, please add it in.

**Comment by MathewBiddle** [comment]
before `git mv` should you `git submodule update --init` and `git submodule update --remote --merge`? or after? Just to catch the repo up?

**Comment by mwengren** [comment]
Yep, the merge broke the navbar in this repo, but not a big deal: https://ioos.github.io/ioos-documentation-jekyll-skeleton/howto.html

I just went ahead and updated master with the submodule path move, and also renamed to 'main' at the same time: https://github.com/ioos/documentation-theme-jekyll/commits/main

Good point about the submodules.  I guess I was assuming once we pushed the GHA file, they'd be updated automagically by GitHub - maybe we can try on one first and see what happens....

**Comment by mwengren** [comment]
One issue we might have to address: forked repositories and reconciling possibly duplicated GHA-triggered commits.  

Might have to wait and see what happens with this the next time the GHA job runs on ioos/ioos-documentation-jekyll-skeleton tomorrow.  

So after making changes to the topnav.html and .gitignore [here](https://github.com/ioos/documentation-theme-jekyll/commit/d958aec9981aacace8c08dc6b2da147afdab9af6), and renaming master to main in this repo, I tried to pull down those changes to my local working copy of  ioos/ioos-documentation-jekyll-skeleton via combination of `git pull` and `git submodule update --init` etc (not sure exactly how the commands I ran were ordered now).

Anyway, the result was that my local working copy was updated to https://github.com/ioos/ioos-documentation-jekyll-skeleton/commit/aa71dcdd66e334e5f5019cb8b316059cba8787fd, which includes https://github.com/ioos/documentation-theme-jekyll/commit/78fc807348bf670c2aaa5d0d01d9c3e5de8dedeb of the `theme` submodule:  (because the GHA hasn't run on that repo yet since I pushed the most recent commit above). 

I then pushed my local working copy back to my fork of the skeleton repo: https://github.com/mwengren/ioos-documentation-jekyll-skeleton, at which point the GHA ran again, and added another commit that then updated the `theme` submodule to: https://github.com/ioos/documentation-theme-jekyll/commit/d958aec9981aacace8c08dc6b2da147afdab9af6.

The good news is that my version of the github.io page is rendering properly with the topnav bar now, so that part worked: 

https://mwengren.github.io/ioos-documentation-jekyll-skeleton/howto.html

The bad news is that I have potentially duplicated GHA commits (or I will tomorrow when the upstream repo updates submodules via cron) - notice the extra commit in the second link:
* https://github.com/ioos/ioos-documentation-jekyll-skeleton/commits/gh-pages
* https://github.com/mwengren/ioos-documentation-jekyll-skeleton/commits/gh-pages

I'm not quite sure how to prevent this from causing merge conflicts later on (or at least some headaches), unless anyone modifying these repos either never forks them, or continually force overwrites their forked `gh-pages` branches on GitHub. 

Or, maybe it wont be an issue.  All of the circular dependencies are definitely headache-inducing though.

**Comment by MathewBiddle** [comment]
> Might have to wait and see what happens with this the next time the GHA job runs on ioos/ioos-documentation-jekyll-skeleton tomorrow.

Note that because I added:
https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/aa71dcdd66e334e5f5019cb8b316059cba8787fd/.github/workflows/sync_theme.yml#L9

You can run the workflow manually from:
https://github.com/ioos/ioos-documentation-jekyll-skeleton/actions/workflows/sync_theme.yml

See here:
![image](https://user-images.githubusercontent.com/8480023/219406222-3e2130cd-4316-43e9-b32d-1069832a1dc8.png)


It will take me a bit to digest the rest of your comment.

**Comment by MathewBiddle** [comment]
I just ran the GHA manually and things seem to have sorted themselves out.
https://ioos.github.io/ioos-documentation-jekyll-skeleton/howto.html

![image](https://user-images.githubusercontent.com/8480023/219408834-f228f29a-2180-4254-bfab-46524417b00a.png)

**Comment by MathewBiddle** [comment]
But, your fork does show that you are 1 commit ahead and 1 commit behind, https://github.com/mwengren/ioos-documentation-jekyll-skeleton - blerg.

Question. Will folks be **Forking** this repo? Or will they be cloning it to their repo of choice?

My process has been to clone one of the doc repos (doesn't really matter which one) and dump the contents into an orhpaned `gh-pages` branch in my existing repo. See bullets 1-5 at https://github.com/ioos/ioos-documentation-jekyll-skeleton/issues/4#issue-1381050875

Not sure how to manage your fork though...

**Comment by mwengren** [comment]
No worries.  It's a handful, it took me awhile to trace commits between repos/submodules to even figure out what happened and I wanted to record it for myself as much as anything else.

Essentially, the issue is duplicated commits on downstream forks.  It seems the [sync_theme job](https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/gh-pages/.github/workflows/sync_theme.yml) runs both on a schedule and [on push](https://github.com/mwengren/ioos-documentation-jekyll-skeleton/actions).   

What happened yesterday was when I pushed my local branch that was current with ioos/ioos-documentation-jekyll-skeleton to my fork at mwengren/ioos-documentation-jekyll-skeleton, the on-push job was triggered that then made [this commit](https://github.com/mwengren/ioos-documentation-jekyll-skeleton/commit/ec172f74242d88d4bcfc92bad0ea5169cbafcb22) that pulled in the most recent upstream submodule changes that weren't yet in ioos/ioos-documentation-jekyll-skeleton.

What's strange is that I don't see the corresponding job that made that commit listed in the [jobs history](https://github.com/mwengren/ioos-documentation-jekyll-skeleton/actions), but that's a mystery for another day.  **Edit:** I understand what's happening now, the GHA job history for mwengren (above) shows that when I pushed the most recent commit [aa71dcd](https://github.com/ioos/ioos-documentation-jekyll-skeleton/commit/aa71dcdd66e334e5f5019cb8b316059cba8787fd) this triggered the job to run, and in the [job log](https://github.com/mwengren/ioos-documentation-jekyll-skeleton/actions/runs/4188459046/jobs/7259618074 ), the 'Commit and push if it changed' step then pushed the resuting new commit [ec172f7](https://github.com/mwengren/ioos-documentation-jekyll-skeleton/commit/ec172f74242d88d4bcfc92bad0ea5169cbafcb22) to the `gh-pages` branch history.  


Then, a [subsequent job](https://github.com/ioos/ioos-documentation-jekyll-skeleton/actions/runs/4195427195/jobs/7274954483) on the main ioos/ioos-documentation-jekyll-skeleton ran and pulled in the same submodule commits (to match my fork).  

But I have now duplicated commits between [my fork](https://github.com/mwengren/ioos-documentation-jekyll-skeleton/commits/gh-pages) and [upstream gh-pages](https://github.com/ioos/ioos-documentation-jekyll-skeleton/commits/gh-pages).

I think we need to adjust the sync_theme) job to introspect the URL of the repository (if possible), and only run if the repo path matches 'https://github.com/ioos/'.  Hopefully that's easy to do.  Can you look into that when you get a chance?

**Comment by mwengren** [comment]
Whoops, missed your earlier comments while writing.  I see you ran the job manually.

Yeah, now I have the commit mis-match issue you mentioned.  Hopefully some extra GHA checks will prevent this from being an issue in any forks.  

 I think using the skeleton repo would be easier than following the process you described in https://github.com/ioos/ioos-documentation-jekyll-skeleton/issues/4#issue-1381050875.  The idea was by using the skeleton you can avoid any submodule config steps.

We might be missing some git-related steps to do that in the [instructions](https://ioos.github.io/ioos-documentation-jekyll-skeleton/howto.html) though.  I could add steps to walk through re-initializing the git references in order to push the cloned repo to a fresh/new repo in GitHub.  Or, we could look into making the skeleton repo a github template, that might help too...

I'd need to do the git repo init steps again to refresh my memory on exactly how to do that, but roughly:
* clone ioos-documentation-jekyll-skeleton 
* rename repo dir
* cd repo dir
* git init (...)
* git push (to your own acct most likely)

Then we migrate from a personal account to a repo under IOOS (at which point it's published to ioos.github.io and GHA takes over maintaining).

**Comment by mwengren** [comment]
Ok, I understand GHA a little better now, edited my https://github.com/ioos/documentation-theme-jekyll/issues/3#issuecomment-1433315947 to describe what happened with my forked repo yesterday with respect to the sync_theme job. 

One more question though, do we need to update the cron config [here](https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/gh-pages/.github/workflows/sync_theme.yml) so it actually runs at Noon (UTC, I guess).  It looks like 10 past 1700  (is that ET)?

**Comment by mwengren** [comment]
Hey @MathewBiddle, I am hoping to get this all resolved ASAP so we can move on to other things hopefully - I've gotten sucked into it at least the past few days.  

I can work on updating the HOWTO page with different approaches to copying our [ioos/ioos-documentation-jekyll-skeleton](https://github.com/ioos/ioos-documentation-jekyll-skeleton) repo for a new documentation site. '

It seems there's two options:
1. the 'fork' approach (as described in the [HOWTO](https://ioos.github.io/ioos-documentation-jekyll-skeleton/howto.html))
1. the 'gh-pages' branch recreate approach you describe in ioos/ioos-documentation-jekyll-skeleton#4

I guess the reason for Option 2 is that you want to add a gh-pages site to an existing repo, and therefore making a full fork is unnecessary and counterproductive, true?   Is there another way you could do this without the branch download/submodule reconfiguration approach?  Could you just clone into ioos-documentation-jekyll-skeleton, checkout the gh-pages branch, and copy the files to the existing local repo?  I haven't tried to do that to know if so.  I don't recall ever trying to import an unrelated branch into an existing repo before.

For Option 1, one step I took to make it easier was to turn the ioos-documentation-jekyll-skeleton repo into a GitHub template, which makes it a little easier to fork.  I'll start by updating the HOWTO with those instructions/images.  

Then, maybe we can figure out how to add Option 2 alongside the 'fork' instructions for Option 1 once we know the easiest approach to do that.

**Comment by MathewBiddle** [comment]
> I am hoping to get this all resolved ASAP so we can move on to other things hopefully - I've gotten sucked into it at least the past few days.

Me too. While it's been fun, I have other more pressing activities to dig into. I think it's in an okay state right now - pending the fork issue - that we could just leave it as is and come back in a week or so...

> I guess the reason for Option 2 is that you want to add a gh-pages site to an existing repo, and therefore making a full fork is unnecessary and counterproductive, true?

True.

>  Is there another way you could do this without the branch download/submodule reconfiguration approach? Could you just clone into ioos-documentation-jekyll-skeleton, checkout the gh-pages branch, and copy the files to the existing local repo?

Found this, will have to give it a go: https://stackoverflow.com/questions/9767381/importing-one-git-repo-as-a-branch-into-another-git-repo

Should I still look into 
> I think we need to adjust the sync_theme) job to introspect the URL of the repository (if possible), and only run if the repo path matches 'https://github.com/ioos/'.

**Comment by MathewBiddle** [comment]
Here's a simple(er?) way to do it

Setup the skeleton repo as a remote for `gh-pages` branch in existing repo:
```
git remote add gh-pages https://github.com/ioos/ioos-documentation-jekyll-skeleton.git
```
Fetch the `gh-pages` content:
```
git fetch gh-pages
```
Checkout `gh-pages`
```
git checkout gh-pages
```
Initialize and pull over the submodules:
```
git submodule update --init
```

Now you're remote for the `gh-pages` branch is still pointing to the skeleton repo:
```
>git remote -v
gh-pages        https://github.com/ioos/ioos-documentation-jekyll-skeleton.git (fetch)
gh-pages        https://github.com/ioos/ioos-documentation-jekyll-skeleton.git (push)
origin  https://github.com/MathewBiddle/surf_check.git (fetch)
origin  https://github.com/MathewBiddle/surf_check.git (push)
```
When we want to push it up, we want to push it to the same repo as `origin`. 

So, set the remote for `gh-pages` branch to the repository you are adding the page to:
```
git remote set-url gh-pages https://github.com/<org>/<repo>.git

>git remote -v                                                               
gh-pages        https://github.com/MathewBiddle/surf_check.git (fetch)
gh-pages        https://github.com/MathewBiddle/surf_check.git (push)
origin  https://github.com/MathewBiddle/surf_check.git (fetch)
origin  https://github.com/MathewBiddle/surf_check.git (push)

```

Now you can push new branch up to GH:
```
git push
```

**Comment by MathewBiddle** [comment]
One option to potentially resolve the conflicts in your fork is to only make changes when the cron job runs or when you manually run the GHA. Ie. Don't run on `push` by removing these lines:
https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/b52aee18970949182c7dd4dd0da8dc394d6248c2/.github/workflows/sync_theme.yml#L4-L6

>One more question though, do we need to update the cron config [here](https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/gh-pages/.github/workflows/sync_theme.yml) so it actually runs at Noon (UTC, I guess). It looks like 10 past 1700 (is that ET)?

Yes I forgot to update the time when I was initially testing. I would prefer it ran in the morning ET so we can tackle any issue that might arise. Would hate for something to happen on a Friday afternoon....

**Comment by MathewBiddle** [comment]
Looking into checking the url of the repo now. See
https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#using-conditionals

**Comment by mwengren** [comment]
> One option to potentially resolve the conflicts in your fork is to only make changes when the cron job runs or when you manually run the GHA. Ie. Don't run on push by removing these lines:
https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/b52aee18970949182c7dd4dd0da8dc394d6248c2/.github/workflows/sync_theme.yml#L4-L6

I think it would make sense to remove the run on push anyway, so yes let's make that change, but...

I think a forked repo will still be a problem no matter what tho if it runs on cron - essentially each fork will have it's own commit making the same submodule updates (each with its own hash).  

Its not hard to drop and re-create the gh-pages branch from upstream if necessary, so this isn't critical.  That's what I did before making further changes in my latest [PR](https://github.com/ioos/ioos-documentation-jekyll-skeleton/pull/8) so I'd be working from the upstream gh-pages commit history.  

I kept my old branch for reference to see how two forks can diverge: https://github.com/mwengren/ioos-documentation-jekyll-skeleton/tree/gh-pages-bak.

But, yes, making it so the
```
git submodule update --init
git submodule update --remote --merge
```
or 
```
git commit -m "Update theme on: ${timestamp}" || exit 0
```
only completes if the repo URL is on https://github.com/ioos/ would be ideal.

As far as the HOWTO, we can add the steps in https://github.com/ioos/documentation-theme-jekyll/issues/3#issuecomment-1434891328 as another option in the Getting Started section instead of the Template/Fork approach once we test it out.

**Comment by MathewBiddle** [comment]
I've got the expression working, just testing on another repo right now
https://github.com/ioos/mbon-docs/blob/94811517d04afd9e0bb9d05307d662ed5e7ac03a/.github/workflows/sync_theme.yml#L8-L12

Running into an issue where the GHA wont run as it sits in the `gh-pages` branch which is not the default. troubleshooting now.

**Comment by MathewBiddle** [comment]
Okay, I have it checking if the repo is owned by IOOS. If it is owned by ioos, it runs. If not, it skips. Below are examples of the two cases.

Owned by me (skipped):
https://github.com/MathewBiddle/surf_check/actions/runs/4207278944/jobs/7301867932

Owned by IOOS (not skipped, although failing because I haven't updated the submodule changes yet):
https://github.com/ioos/mbon-docs/actions/runs/4207296613/jobs/7301909303

Both are the same GHA.

**Comment by MathewBiddle** [comment]
The last problem we have is that the GHA must be in the default branch if we remove 
```
on:
  push:
    branches: gh-pages
```

So, we either include that the user needs to make `gh-pages` default in the docs. Or, we leave the `on: push:` in GHA. I vote for the later.

**Comment by mwengren** [comment]
Ok, that looks like a good change as far as using the conditional for repo ownership.  I'm still learning about GHA, but reading up on the docs now.  

> The last problem we have is that the GHA must be in the default branch if we remove 
```
on:
  push:
    branches: gh-pages
```

I guess I don't understand why it must be in the default branch without the above `on - push` stuff.  Won't it still run on the schedule defined, as in the docs [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule)?  Here's the relevant example that shows that configuration, which matches what you have:
```
on:
  schedule:
    - cron: '30 5 * * 1,3'
    - cron: '30 5 * * 2,4'
```

For some of the repos, the default branch is `gh-pages`, although not all of them (ioos.github.io is one example).

**Comment by MathewBiddle** [comment]
Maybe the `schedule` will run while not being default (just have to wait that out). But the `workflow_dispatch` will not run (that allows us to manually run the action).

**Comment by MathewBiddle** [comment]
Note, I ran into this error when running `git submodule update --remote --merge` locally for mbon-docs (which hasn't been updated with the changes we've made.).
```
warning: ignoring dangling symref refs/remotes/origin/HEAD
fatal: Needed a single revision
Unable to find current origin/HEAD revision in submodule path 'theme'
```

According to this https://stackoverflow.com/questions/45811971/warning-ignoring-broken-ref-refs-remotes-origin-head need to do some fiddling with the symbolic refs...
```
git symbolic-ref refs/remotes/origin/HEAD refs/remotes/origin/main
```

**Comment by MathewBiddle** [comment]
I don't know how to fix that last piece. Will probably delete and re-clone just to move on with it. Seems to have worked though and was able to push up the changes to `navbars` and `theme`. Just got that odd warning and fatal notice.

**Comment by mwengren** [comment]
Is that while running in your local clone, or during the GHA job?  

I think it could be related to the master -> main branch change in https://github.com/ioos/documentation-theme-jekyll, but not sure.  

I ran a few commands GitHub showed on the web UI after I made the change, but that was in my local clone of documentation-theme-jekyll, not in the submodule:
```
git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
```
I'm not sure if I've done the `git submodule update --remote --merge` command since making the change or not.

**Comment by MathewBiddle** [comment]
> Is that while running in your local clone, or during the GHA job?

Yes, I was updating my local version and it was because of the master -> main change. Simple enough to delete and reclone and error goes away. It has something to do with the references in the .git config, but didn't impact anything. 

I think we can call this effort done. 

Do we need to merge in https://github.com/ioos/ioos-documentation-jekyll-skeleton/compare/submodule-rename-2...gh-pages Or, can we delete that branch?

**Comment by MathewBiddle** [comment]
I think that was part of this PR https://github.com/ioos/ioos-documentation-jekyll-skeleton/pull/7

**Comment by mwengren** [comment]
I just deleted it.  That was a user error on my part not checking where I pushed it originally, I meant to push to my fork for the PR and pushed it there instead.  

I'm going to take another look in a bit and hopefully wrap up a couple lingering things before calling this done.

**Comment by mwengren** [comment]
I think it's looking pretty good for now.  In addition to the PR: https://github.com/ioos/ioos-documentation-jekyll-skeleton/pull/10 I just sent, we need to:

- [ ] run these steps https://github.com/ioos/documentation-theme-jekyll/issues/3#issuecomment-1431999268  on each doc repository I think.  
- [ ] update [navbars](https://github.com/ioos/documentation-theme-jekyll/tree/navbars) branch with changes: https://github.com/ioos/documentation-theme-jekyll/issues/2#issuecomment-1416189136 (and any other changes we want to make) to test that everything's working

**Comment by mwengren** [comment]
Here's a list of documentation repos I know of.  If we divide 50/50 and update each with the [steps above](https://github.com/ioos/documentation-theme-jekyll/issues/3#issuecomment-1431999268), does that sound ok?  It might be scriptable, but I'd probably opt to just do it manually...

Matt:

- [x] BioData-Training-Workshop
- [x] biological-data-services (+ add deprecation notice)?
- [x] documentation-theme-jekyll
- [x] conventions-for-observing-asset-identifiers
- [x] mbon-docs
- [x] ncei-archiving-cookbook
- [x] erddap-gold-standard

Micah:

- [x] ioos-csv-tsv (+ add deprecation notice)
- [x] ioos-documentation-jekyll-skeleton
- [x] ioos.github.io
- [x] ioos-metadata
- [x] ioosngdac
- [x] sos-guidelines  (+ add deprecation notice)
- [x] passive-acoustics
- [x] configuration-management (need to unarchive repo and turn off GH Pages site, if possible)

Anything missing?

I'll try to update mine over the next few days.

**Comment by MathewBiddle** [comment]
That list looks good to me. I'll start working on it today.

I think we might be able to get away with using [VS code in browser](https://visualstudiomagazine.com/articles/2021/08/31/github-vs-code.aspx) to make these changes.  (Trying to limit clone local steps.) I'll try it out.

**Comment by MathewBiddle** [comment]
I don't think we need this as it was a change to the theme itself.

> M _includes/topnav.yml - [line 28](https://github.com/ioos/documentation-theme-jekyll/blob/6674af5f1b065d2d24e358472f428f864b023c60/_includes/topnav.html#L28)

**Comment by MathewBiddle** [comment]
Also need to:
> M .gitmodules [line 5](https://github.com/ioos/ioos-documentation-jekyll-skeleton/blob/dd81c9e03ae9272b7c2e4bf42107b29e956db6b9/.gitmodules#L5)

**Comment by MathewBiddle** [comment]
Okay, here's what I did:

https://github.com/ioos/ncei-archiving-cookbook/commit/d4a848494efc4a809a6d8117c1eccf9ce31f805c

But I couldn't get VS Code (in browser) to register the rename of `_data/navbars`. So, I ended up cloning the repo locally and running
> git mv _data/navbars_theme _data/navbars

https://github.com/ioos/ncei-archiving-cookbook/commit/70d7f000d656edd0a320903e4f739fe8c8ec24ba

Edits to `.gitmodules` happens when you run `git mv`.

**Comment by MathewBiddle** [comment]
Here are all the steps in one commit:
https://github.com/ioos/BioData-Training-Workshop/commit/4ae68c502874166735106e7472921e80f2a7d5df

I'm sure we can shell script this out too ;)

**Comment by mwengren** [comment]
Yes, `git mv _data/navbars_theme _data/navbars` covers the change to `.gitmodules` from my testing on the skeleton, so shouldn't need to edit `.gitmodules` directly at all.  

I already have all the repos cloned locally, so I'll just do it that way.  Maybe I'll give VSCode in browser a try sometime, new to me...

**Comment by MathewBiddle** [comment]
Since I have to clone them, I cobbled together this shell script:

```bash
#! /bin/bash

REPO=conventions-for-observing-asset-identifiers

git clone https://github.com/ioos/$REPO.git

cd $REPO

git checkout -b sync-theme gh-pages

git status

mkdir -p .github/workflows

touch .github/workflows/sync_theme.yml

echo 'name: sync theme submodules

on:
  push:
    branches: gh-pages

  schedule:
     - cron: "00 14 * * *"

  workflow_dispatch:


jobs:
  sync_theme_submodules:
    if: github.repository_owner == '\''ioos'\'' # only run if ioos owns repo
    runs-on: ubuntu-20.04
    steps:
    - name: Checkout repo
      uses: actions/checkout@v3

    - name: submodule checkout
      run: |
           git submodule update --init
    - name: check submodule status
      run: |
          git submodule status
    - name: Update theme from submodules
      run: |
           git submodule update --remote --merge
    - name: check submodule status 2
      run: |
           git submodule status
    - name: Commit and push if it changed
      run: |
           git config user.name "Automated"
           git config user.email "actions@users.noreply.github.com"
           git add -A
           timestamp=$(date -u)
           git commit -m "Update theme on: ${timestamp}" || exit 0
           git push' > .github/workflows/sync_theme.yml

sed -i 's/.jekyll-metadata/.jekyll-metadata\n.ipynb_checkpoints/g' .gitignore

git mv _data/navbars_theme _data/navbars

git status

git add .gitignore .github/

git rm _data/topnav.yml

git status

git commit -m "following ioos/documentation-theme-jekyll#3"

git status
```
Outside the script, `cd` to the repo and `git push` (this could be added to the script above, but I wanted to manually check before I pushed up the new branch.)
```command
 git push --set-upstream origin sync-theme
```

**Comment by MathewBiddle** [comment]
I've also been adding a link to the site in the GH description:
![image](https://user-images.githubusercontent.com/8480023/220973480-71060c20-9c64-42fc-9bd2-49de8cb51850.png)

**Comment by mwengren** [comment]
Not sure if it matters, I guess we'll see if it runs tomorrow morning, but since the [ioos/ioos.github.io](https://github.com/ioos/ioos.github.io) repo uses the `main` branch by convention rather than `gh-pages`, the GHA didn't run when I merged my PR just now.  See:

https://github.com/ioos/ioos.github.io/actions

**Comment by mwengren** [comment]
A couple more things to note:

Probably not necessary, but I added a _config_dev.yml file for each of the repos that was missing it.  I did for the ioos/ioos-csv-tsv repo as part of the PR and then forgot to do it for the others, so I just pushed individual commits with it this morning to `gh-pages` branches.

If you do the same, you might run into the issue I had with pulling in the automated GHA commits with submodule changes.

The steps I wrote in the docs here: https://ioos.github.io/ioos-documentation-jekyll-skeleton/howto.html#a-note-on-submodules aren't entirely accurate.  

If I run the following to try to update my local clone, I need to run both
```
git pull
git submodule update --init
```
To pull in the submodule updates.  Running `git pull` alone is not sufficient.

If the latest submodule commits aren't pulled in as well, if you make further changes it can actually revert back to a previous submodule commit, effectively undoing the changes made in the GHA commit.   I made this mistake once and had to undo my changes when I realized what was going on.

I'm not sure actually what the `git submodule update --remote --merge` step is for now, because I had the best results by running the two commands above on each cloned repo before making any new changes.  At least, for this particular workflow, it wasn't necessary.

If you add a `_config_dev.yml` file to the repos you were working with (if not already there), I'd be curious how it goes for you and whether you find the same.  

I might update the HOWTO docs again with the approach above to clarify - currently it says you can just do `git pull`, which is incorrect (I assumed that would work so I added those instructions in).

This shows what I'm talking about.  If you were to commit changes after the first `git status` command, you would reset the submodules to an earlier commit.

```console
git checkout gh-pages 
warning: unable to rmdir _data/navbars: Directory not empty
Switched to branch 'gh-pages'
Your branch is behind 'origin/gh-pages' by 2 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
(base) mwengren@kachina:~/workspace/code/git/ioos/passive-acoustics$ git pull
Updating b719f53..4ff9d67
Fast-forward
 .github/workflows/sync_theme.yml | 40 ++++++++++++++++++++++++++++++++++++++++
 .gitignore                       |  1 +
 .gitmodules                      |  2 +-
 _data/navbars                    |  1 +
 _data/navbars_theme              |  1 -
 _data/topnav.yml                 | 52 ----------------------------------------------------
 theme                            |  2 +-
 7 files changed, 44 insertions(+), 55 deletions(-)
 create mode 100644 .github/workflows/sync_theme.yml
 create mode 160000 _data/navbars
 delete mode 160000 _data/navbars_theme
 delete mode 100644 _data/topnav.yml
(base) mwengren@kachina:~/workspace/code/git/ioos/passive-acoustics$ git status                                                                                                                         
On branch gh-pages
Your branch is up-to-date with 'origin/gh-pages'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   _data/navbars (new commits)
        modified:   theme (new commits)

no changes added to commit (use "git add" and/or "git commit -a")
(base) mwengren@kachina:~/workspace/code/git/ioos/passive-acoustics$ git submodule status
+248d4c8ac0091d4f36b92ee39e45903e9fcd6d8e _data/navbars (remotes/origin/revert-24-gh-pages-158-g248d4c8)
+6674af5f1b065d2d24e358472f428f864b023c60 theme (remotes/origin/HEAD)
(base) mwengren@kachina:~/workspace/code/git/ioos/passive-acoustics$ git submodule update --init
Submodule path '_data/navbars': checked out 'edb2048843b71191909d53428b969f53bfbe4757'
Submodule path 'theme': checked out 'd958aec9981aacace8c08dc6b2da147afdab9af6'
(base) mwengren@kachina:~/workspace/code/git/ioos/passive-acoustics$ git submodule status                                                                                                               
 edb2048843b71191909d53428b969f53bfbe4757 _data/navbars (remotes/origin/navbars)
 d958aec9981aacace8c08dc6b2da147afdab9af6 theme (remotes/origin/main)
(base) mwengren@kachina:~/workspace/code/git/ioos/passive-acoustics$ git status
On branch gh-pages
Your branch is up-to-date with 'origin/gh-pages'.
nothing to commit, working directory clean

```

**Comment by mwengren** [comment]
Whoops, googling 'IOOS DMAC documentation' led me to this link/repo:  https://ioos.github.io/configuration-management/

Somehow this one slipped through the cracks - I don't think we ever added it to our menu bar and as a result it became sort of orphaned.  You can see a much earlier version of the top nav bar on it as well.  

I don't think this guidance is relevant any longer, so we should at least add a deprecation notice to the content pages.  

This dates back to @dpsnowden's time as DMAC System Architect, he might enjoy seeing it online still (or maybe not...).  If you see this Derrick, how should we handle this material?

Update: I see we/I archived the repo several years ago - https://github.com/ioos/configuration-management, but must have forgotten that the site would still be online.  Easiest thing is probably to disable the GitHub Pages site for it, if possible, unless we wanted to retain it for some reason.

**Comment by mwengren** [comment]
Update: https://ioos.github.io/configuration-management/ has been unpublished.

**Comment by MathewBiddle** [comment]
Can we put a fork in this? 🍴

**Comment by mwengren** [comment]
I think the only loose end I wanted to fix was updating the HOWTO with correct steps to update submodules in local clones with upstream GHA-originated commits, and preventing unintentional commit reverts down the road.  

I'm sure I'll forget how to do this properly so I wanted to be sure the steps in the docs were right.

Have you tried doing this yet?  See https://github.com/ioos/documentation-theme-jekyll/issues/3#issuecomment-1443908757.

**Comment by MathewBiddle** [comment]
I have not. But, I need to update the ioos-atn-data page with the action, so I'll give it a go. https://github.com/ioos/ioos-atn-data/issues/76

**Comment by mwengren** [comment]
Didn't realize that one existed either - we should add that to the topnav header per: https://github.com/ioos/documentation-theme-jekyll/issues/2#issuecomment-1416189136.  I plan to make those changes once this issue is closed and we can use it as a test case for the automation.  I might pull that comment out into a separate issue to track it better.

As far as the submodule stuff goes, you'll run into that on any of the existing repos when you do a git pull on a local clone, since it'll bring in the GHA submodule update commits.  I'll test a few more myself today to confirm I have the steps right.

**Comment by MathewBiddle** [comment]
> As far as the submodule stuff goes, you'll run into that on any of the existing repos when you do a git pull on a local clone, since it'll bring in the GHA submodule update commits. I'll test a few more myself today to confirm I have the steps right.

*If your local clone's submodule is behind.

Yes, luckily I had one that was behind.

```
$ git status
On branch gh-pages
Your branch is up to date with 'origin/gh-pages'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        _data/navbars/

nothing added to commit but untracked files present (use "git add" to track)

$ git submodule status
-edb2048843b71191909d53428b969f53bfbe4757 _data/navbars_theme
 6674af5f1b065d2d24e358472f428f864b023c60 theme (heads/master)

$ git fetch
remote: Enumerating objects: 38, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (14/14), done.
remote: Total 24 (delta 10), reused 14 (delta 6), pack-reused 0
Unpacking objects: 100% (24/24), 7.43 KiB | 111.00 KiB/s, done.
From https://github.com/ioos/ioos-documentation-jekyll-skeleton
   78ffea8..dd81c9e  gh-pages   -> origin/gh-pages
Fetching submodule theme
From https://github.com/ioos/documentation-theme-jekyll
 * [new branch]      main       -> origin/main

$ git pull
Updating 78ffea8..dd81c9e
Fast-forward
 .github/workflows/sync_theme.yml | 58 ++++++++++++------------
 .gitmodules                      |  2 +-
 README.md                        | 47 +++++++++++++++++---
 _data/{navbars_theme => navbars} |  0
 _docs/howto.md                   | 95 ++++++++++++++++++++++++----------------
 theme                            |  2 +-
 6 files changed, 130 insertions(+), 74 deletions(-)
 rename _data/{navbars_theme => navbars} (100%)

$ git status
On branch gh-pages
Your branch is up to date with 'origin/gh-pages'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   theme (new commits)

no changes added to commit (use "git add" and/or "git commit -a")

$ git submodule update --init
Submodule path 'theme': checked out 'd958aec9981aacace8c08dc6b2da147afdab9af6'

$ git submodule status
 edb2048843b71191909d53428b969f53bfbe4757 _data/navbars (remotes/origin/navbars)
 d958aec9981aacace8c08dc6b2da147afdab9af6 theme (remotes/origin/main)
```

Even if you push the previous submodule commit without `git submodule update --init`, the GHA will update to the most recent version. 

It might look funky in the commit history (dropping the submodules back one, then up one), the published page will be correct. 

I like your thought to add `git submodule update --init` to the process of working on a page.

**Comment by MathewBiddle** [comment]
My understanding is

```
git submodule update --init
```
Pulls in the submodules with the last commit they where established under (might not be the most up-to-date)

```
git submodule update --remote --merge
```
Checks the submodules for any new commits to their appropriate repo/branch and brings the changes over.

**Comment by mwengren** [comment]
It's messy to undo submodule commits only to have GHA redo them, but I think I have better instructions now that should help - I've been tinkering with this for awhile but understand better now.  Or, at least I found a workflow that works for me when I try it.  

Basically, just:

```
git pull
git submodule update
```

Leaving out the `--remote --merge` flags.  The `--remote` flag apparently uses 'the status of the submodule’s remote-tracking branch', rather than 'using the superproject’s recorded SHA-1 to update the submodule', which is bad.  

The docs are here: https://git-scm.com/docs/git-submodule, but honestly they're pretty confusing.  

`--merge` is actually a different update strategy entirely (when omitted the update strategy defaults to 'checkout' strategy) which is what we want because it uses the superproject commit history to pull the correct commit out for the submodules.

It doesn't appear to matter in this case if you add `--init` at the end or not, so I just included both in the docs:

```
git submodule update 
git submodule update --init
```

Anyway, I have a PR I'll submit just about ready.

**Comment by mwengren** [comment]
@MathewBiddle I think we can close this finally!  

I've tested the workflow in https://ioos.github.io/ioos-documentation-jekyll-skeleton/howto.html#a-note-on-submodules a few times and it's pretty reliable to pull in the upstream GHA-originated submodule commits to a local clone.  

I think the next step here is updating the menu bar.  New issue created: https://github.com/ioos/documentation-theme-jekyll/issues/10.

Thanks for all your help with this!

**Comment by MathewBiddle** [comment]
@mwengren thanks for all your help! Great team effort.
